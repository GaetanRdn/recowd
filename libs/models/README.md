# models

## description

Contains all models defined in the workspace.
