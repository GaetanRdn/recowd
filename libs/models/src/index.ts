export * from './ui/contract-type';
export * from './ui/education';
export * from './ui/experience';
export * from './ui/partner';
export * from './ui/role';
