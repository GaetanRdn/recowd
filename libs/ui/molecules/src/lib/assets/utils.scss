@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
