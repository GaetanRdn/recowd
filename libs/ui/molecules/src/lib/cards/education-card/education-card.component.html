<rc-card>
  <img
    *ngIf="education.logo"
    alt="logo"
    rcAvatar
    src="assets/{{ education.logo }}"
  />
  <rc-card-title>{{ education.degree }}</rc-card-title>
  <rc-card-subtitle>{{ education.from }} - {{ education.to }}</rc-card-subtitle>
  <rc-card-content>{{ education.description }}</rc-card-content>
</rc-card>
