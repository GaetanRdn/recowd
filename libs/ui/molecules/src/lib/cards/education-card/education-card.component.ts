import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AvatarDirective, CardModule } from '@recowd/ui-atoms';
import { Required } from '@recowd/utility-types';
import { Education } from '@recowd/models';

@Component({
  selector: 'rc-education-card',
  standalone: true,
  imports: [CommonModule, CardModule, AvatarDirective],
  templateUrl: './education-card.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class EducationCardComponent {
  @Input() @Required() public education!: Education;
}
