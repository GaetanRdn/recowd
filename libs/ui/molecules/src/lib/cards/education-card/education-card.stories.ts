import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { EducationCardComponent } from './education-card.component';

export default {
  title: 'molecules/cards/education-card',
  decorators: [
    moduleMetadata({
      imports: [EducationCardComponent],
    }),
  ],
} as Meta<EducationCardComponent>;

const Template: Story<EducationCardComponent> = (
  args: EducationCardComponent
) => ({
  props: {
    education: args.education,
  },
  template: `<rc-education-card [education]="education"></rc-education-card>`,
});

export const Default = Template.bind({});
Default.args = {
  education: {
    degree: 'Master MIAGE par apprentissage',
    from: 2012,
    to: 2014,
    description:
      "Université Paris-Sud, Méthodes Informatique Appliquées à la Gestion d'entreprise",
  },
};
