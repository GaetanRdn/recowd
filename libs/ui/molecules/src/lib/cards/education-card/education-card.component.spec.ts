import { TestBed, waitForAsync } from '@angular/core/testing';

import { EducationCardComponent } from './education-card.component';
import { TemplateLookup } from '@recowd/test/utils';
import { Education } from '@recowd/models';
import { Component } from '@angular/core';

describe('EducationCardComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [EducationCardComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
  }));

  it.each([
    {
      from: '2021',
      to: '2022',
      degree: 'Master',
      logo: undefined,
      description: 'long comment',
    },
    {
      from: 2018,
      to: 2019,
      degree: 'Bac',
      logo: 'logo.png',
      description: 'another comment',
    },
  ] as Education[])('should create with %j', (education: Education) => {
    templateLookup.hostComponent.education = education;
    templateLookup.detectChanges();
    expect(templateLookup.fixture.nativeElement.children).toMatchSnapshot();
  });
});

@Component({
  template: ` <rc-education-card [education]="education"></rc-education-card>`,
})
class HostComponent {
  public education!: Education;
}
