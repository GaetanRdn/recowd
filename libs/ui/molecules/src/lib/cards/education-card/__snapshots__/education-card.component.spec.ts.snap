// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EducationCardComponent should create with {"from":"2021","to":"2022","degree":"Master","description":"long comment"} 1`] = `
HTMLCollection [
  <rc-education-card>
    <rc-card>
      <div
        class="rc-card-header"
      >
        
        <rc-card-title
          class="rc-card-title"
        >
          Master
        </rc-card-title>
        <rc-card-subtitle
          class="rc-card-subtitle"
        >
          2021 - 2022
        </rc-card-subtitle>
      </div>
      <rc-card-content
        class="rc-card-content"
      >
        long comment
      </rc-card-content>
    </rc-card>
  </rc-education-card>,
]
`;

exports[`EducationCardComponent should create with {"from":2018,"to":2019,"degree":"Bac","logo":"logo.png","description":"another comment"} 1`] = `
HTMLCollection [
  <rc-education-card>
    <rc-card>
      <div
        class="rc-card-header"
      >
        <img
          alt="logo"
          class="rc-avatar"
          rcavatar=""
          src="assets/logo.png"
        />
        
        <rc-card-title
          class="rc-card-title"
        >
          Bac
        </rc-card-title>
        <rc-card-subtitle
          class="rc-card-subtitle"
        >
          2018 - 2019
        </rc-card-subtitle>
      </div>
      <rc-card-content
        class="rc-card-content"
      >
        another comment
      </rc-card-content>
    </rc-card>
  </rc-education-card>,
]
`;
